### ISSUE TITLE
Make it spawn regular Settler unit instead of Settler replacament

### ISSUE THREAD

**Issue by ravignir** [yairm210/Unciv#4757]
Make one of the starting settlers unit from eras.json a regular settler, not a civ replacament unique unit. This is important for the mods that add nations that can't settle more cities, bebause their settler replacament lacks "Founds a new City" unique. Playing as these nations is currently impossible, because game is automatically lost/they can't settle capital city.

Other than that: it is also what happens in civ5; Venice in later eras spawn with just one settler and couple of merchants of venice.

Another issue that is kind of related, is the fact some civs should not spawn with their unique units - like the huns, who spawn with warriors and scouts instead of battering rams in civ5.

**Comment by yairm210** [comment]
If this is related to specific units, we should probably add a "Does not replace starting unit" unique to these

**Comment by logicminimal** [comment]
> If this is related to specific units, we should probably add a "Does not replace starting unit" unique to these

i think we should also a unique add settlers to receive in the start of the game

**Comment by SomeTroglodyte** [comment]
Aren't battering rams a replacement for spearmen not warriors? Then Huns with warriors would be perfectly normal??

**Comment by xlenstra** [comment]
If I'm not mistaken, Ravignir here talks about starting in the classical era. If you do that with the huns in Civ V, you receive a warrior and a scout, while other civs start with two spearman instead.

On the issue itself: what about always making sure that if units are replaced, at least one of them still has the "can found cities" unique? Possibly with a mod-option for disabling this if you for some reason want to create a mod without cities. That would solve the issue here, and also make sure venice-like civs only start on later eras with a single settler with the rest replaced with merchants of Venice or some other unit, which the 'does not replace starting unit' unique would not.

**Comment by SomeTroglodyte** [comment]
Venice will need a solution to start properly in _any_ age, even 4000BC if I am not mistaken. That will sort itself out - no, will be sorted out by the masochist taking on Venice.

**Comment by ravignir** [comment]
>If I'm not mistaken, Ravignir here talks about starting in the classical era. If you do that with the huns in Civ V, you receive a warrior and a scout, while other civs start with two spearman instead.
>
Yup, that's what happens. Otherwise their closest neighbours would have a big problem. In medieval era, they start with two warriors and one scout.

**Comment by SomeTroglodyte** [comment]
Well, then it's clear more data in json is needed...

```json
		// The Huns
		"uniques": [
			"[+1 Production] from every [Pasture]", "Cities are razed [2] times as fast", "Starts with [Animal Husbandry]", "\"Borrows\" city names from other civilizations in the game",
			"Receive [1] [Scout] when starting in [Classical era]",
			"Receive [1] [Warrior] when starting in [Classical era]",
			"Receive [0] [Battering Ram] when starting in [Classical era]",
			...
		],
```
... ugly :vomiting_face:

**Comment by SomeTroglodyte** [comment]
maybe that would look better when using eras.json and giving _them_ uniques - but that's my point: decide carefully which primary dimension we want to attach the exceptions data to. Maybe inspect a complete list of exceptions first? So far I know of the Huns and Venice... Too lazy to research.

**Comment by ravignir** [comment]
The Huns:
Classical: Warrior+Scout
Medieval: Warior+Warrior+Scout

Iroquois:
Medieval: Spearman+Mohawk Warrior+Mohawk Warrior

Shoshone:
Up to Rrenaissance they always start with one Pathfinder instead of one of their military units.

Venice:
All of their settlers but one, are replaced with Merchants of Venice.

**Comment by ravignir** [comment]
Oh, forgot about Babylon: They get their Scientist when they start in later eras.

**Comment by logicminimal** [comment]
i think we should add starters to nation jsons instead of hardcoding all of them into getting many of the same powerful military unit

like for Venice in many alter eras (those starting units are random)
"settler": 1
"merchant of venice": "2"

etc i think yall got the pount

**Comment by ravignir** [comment]
So, anything? I have the same problem with Epyrus in my ancient civs mod which has a settler replacement unit that can't found a city.

**Comment by logicminimal** [comment]
> So, anything? I have the same problem with Epyrus in my ancient civs mod which has a settler replacement unit that can't found a city.

as a temporary fix what about making that normal settler only be purchasable with faith and it would cost something like 999999 faith?

**Comment by ravignir** [comment]
Nope, because you can get a second settler from liberty.

**Comment by SomeTroglodyte** [comment]
> So, anything? I have the same problem with Epyrus in my ancient civs mod...

Well, right now I would think
- This is a problem in three dimensions: Nation, Era, Difficulty.
- The cardinality is thus order of magnitude a few hundred, of which a few handfuls of cells need data.
- The data at each node is a collection of unit names (a count seems avoidable by repetition), which would _not_ get a nation-unique mapping, and would override all of the starting units defined by Era.
- Doing it without a tabular exceptions structure would require simple rules, and I can't see them - the _kind_ of logic for each of the listed nations seems distinct. So I'd prefer a sparse N-dimensional exception matrix as just hinted at.
- Data is only mostly complete: Your list ignores the Difficulty axis - and I doubt the Iroquois get an exception only in a Medieval start but not earlier or later eras.
- [Difficulties](https://github.com/yairm210/Unciv/blob/master/android/assets/jsons/Civ%20V%20-%20Vanilla/Difficulties.json) come with extra units which do get nation-unique mapping treatment, in three properties two of which are unused in Vanilla. The longest entry is for Deity, `"aiMajorCivBonusStartingUnits": ["Settler", "Worker", "Scout", "Era Starting Unit", "Era Starting Unit", "Era Starting Unit"]`.
- It _may_ be just right to take the difficulty axis out of the problem and add those units like they are now, _mapped_ for all civs on top of the era/exceptions result. Yes or no? I'd guess no, too OP in some cases.
- So where do we put a conceptual table with high cardinality over a three-part primary key and very sparse population? The sparsity would be a point in favour of uniques, but those wold get way ugly quickly, while a new field Nation.startingUnits could be made to look much cleaner in json. Needs a consensus before anybody starts coding.
- Last not least: GameStarter.addCivStartingUnits is a monster of 146 lines, 224 counting functions only used by it. And it looks hellishly inefficient. I couldn't touch it without succumbing to doing a complete redesign including half of the map editor thanks to the ill-advised "Startinglocation" not-improvements...

**Comment by yairm210** [comment]
Yeah, this is definitely a wontfix.
Settler replacements can be replaced with 1 settler, solving the original problem.
Everything else is out of scope.
